UNITTESTS/stubs/Mutex_stub.cpp

```diff
@@ -27,7 +27,7 @@ rtos::Mutex::~Mutex()
     return;
 }
 
-osStatus rtos::Mutex::lock(unsigned int)
+osStatus rtos::Mutex::lock(void)
```

Comment:
@0xc0170 Since unit tests use `UNITTESTS/stubs/Mutex_stub.cpp` to implement `rtos/Mutex.h` and rtos::Mutex::lock() was not implemented, a test failed with:
```
Undefined symbols for architecture x86_64:

  "rtos::Mutex::lock()", referenced from:

      InternetSocket::open(NetworkStack*)       in libfeatures_netsocket_UDPSocket.MbedOS.a(InternetSocket.cpp.o)

      InternetSocket::close()      in libfeatures_netsocket_UDPSocket.MbedOS.a(InternetSocket.cpp.o)

      InternetSocket::bind(SocketAddress const&)  in libfeatures_netsocket_UDPSocket.MbedOS.a(InternetSocket.cpp.o)

      InternetSocket::set_timeout(int)    in libfeatures_netsocket_UDPSocket.MbedOS.a(InternetSocket.cpp.o)

      InternetSocket::setsockopt(int, int, void const*, unsigned int) in libfeatures_netsocket_UDPSocket.MbedOS.a(InternetSocket.cpp.o)

      InternetSocket::getsockopt(int, int, void*, unsigned int*) in libfeatures_netsocket_UDPSocket.MbedOS.a(InternetSocket.cpp.o)

      InternetSocket::sigio(mbed::Callback<void ()>)    in libfeatures_netsocket_UDPSocket.MbedOS.a(InternetSocket.cpp.o)

      ...

ld: symbol(s) not found for architecture x86_64

collect2: error: ld returned 1 exit status
```

Reply:
Now it implements both.